cmd/cosign/cli/sign/sign_blob.go

```diff
@@ -98,18 +99,15 @@ func SignBlobCmd(ro *options.RootOptions, ko options.KeyOpts, regOpts options.Re
 		signedPayload.Bundle = cbundle.EntryToBundle(entry)
 	}
 
-	// if bundle is specified, just do that and ignore the rest
-	if ko.RFC3161TimestampPath != "" {
-		signedPayload.Base64Signature = base64.StdEncoding.EncodeToString(sig)
-
-		contents, err := json.Marshal(signedPayload)
+	if ko.RFC3161TimestampPath != "" && rfc3161Timestamp != nil {
+		ts, err := json.Marshal(rfc3161Timestamp)
 		if err != nil {
 			return nil, err
 		}
-		if err := os.WriteFile(ko.RFC3161TimestampPath, contents, 0600); err != nil {
+		if err := os.WriteFile(ko.RFC3161TimestampPath, ts, 0600); err != nil {
```

Comment:
Not to self, should require that both TSAServerURL and RFC3161TimestampPath is set